magefile.go

```diff
@@ -0,0 +1,287 @@
+	fmt.Println("Running operator binary build")
+	out, err := goBuild("bin/operator", "./operator")
+	if err != nil {
+		return err
+	}
+	fmt.Println("Finished building operator")
+	fmt.Println("Command Output: ", out)
+
+	return nil
+}
+
+// Generate Go build of the interceptor binary
+func BuildInterceptor(ctx context.Context) error {
+	fmt.Println("Running interceptor binary build")
+	out, err := goBuild("bin/interceptor", "./interceptor")
+	if err != nil {
+		return err
+	}
+	fmt.Println("Finished building interceptor")
+	fmt.Println("Command Output: ", out)
+
+	return nil
+}
+
+// Build all binaries
+func BuildAll() {
+	fmt.Println("Building all binaries")
+	mg.Deps(BuildScaler, BuildOperator, BuildInterceptor)
+}
+
+// --- Docker --- //
+
+// Builds a docker image specified by the name argument with the repository prefix
+func DockerBuild(ctx context.Context, repository string, name string) error {
+	if err := isValidModule(name); err != nil {
+		return err
+	}
+
+	fmt.Println(fmt.Sprintf(
+		"Running docker build for image %q",
+		getFullImageName(repository, name),
+	))
+
+	err := sh.RunV(
+		"docker",
+		"build",
+		"-t",
+		getFullImageName(repository, name),
+		"-f",
+		fmt.Sprintf("%s/Dockerfile", name),
+		".",
+	)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(fmt.Sprintf("Finished building %q", getFullImageName(repository, name)))
+	return nil
+}
+
+// Pushes a given image name to a given repository
+func DockerPush(ctx context.Context, repository string, name string) error {
+	if err := isValidModule(name); err != nil {
+		return err
+	}
+
+	fmt.Println(fmt.Sprintf(
+		"Running docker push for image %q",
+		getFullImageName(repository, name),
+	))
+
+	err := sh.RunV(
+		"docker",
+		"push",
+		getFullImageName(repository, name),
+	)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(fmt.Sprintf("Finished pushing %q", getFullImageName(repository, name)))
+	return nil
+}
+
+// Builds all the images to the given repository
+func DockerBuildAll(repository string) {
+	var fns []interface{}
+	for _, module := range []ModuleName{SCALER, OPERATOR, INTERCEPTOR} {
+		fns = append(fns, mg.F(DockerBuild, repository, (string)(module)))
+	}
+	mg.Deps(fns...)
+}
+
+// Pushes all the images to the given repository
+func DockerPushAll(repository string) {
+	var fns []interface{}
+	for _, module := range []ModuleName{SCALER, OPERATOR, INTERCEPTOR} {
+		fns = append(fns, mg.F(DockerPush, repository, (string)(module)))
+	}
+	mg.Deps(fns...)
+}
+
+// --- Helm --- //
+
+// Upgrades or installs the Add-on onto the current cluster.
+// Issuing "mage helmupgradeoperator kedahttp kedacore" will download
+// "kedacore/http-add-on-operator:{SHA}" image and install along with the
+// interceptor and scaler images on the same SHA
+func UpgradeOperator(namespace string, imageRepository string) error {
+	if namespace == "" {
+		namespace = DEFAULT_NAMESPACE
+	}
+
+	if err := sh.RunV(
+		"helm",
+		"upgrade",
+		DEFAULT_NAMESPACE,
```

Comment:
shouldn't this be the helm chart name instead?

Reply:
Yes, the default namespace and the chart name are the same, I must have changed it as a bulk edit. Will correct it

Reply:
Changed on 6e9629d

Reply:
Updated on 6e9629d